src/app/games/martian-chess/martian-chess.component.html

```diff
@@ -0,0 +1,100 @@
+        <text *ngIf="countDown.isPresent()"
+              id="countDownText"
+              class="text-125 text-center"
+              [ngClass]="'player' + getCurrentPlayer().value"
+              [attr.x]="0.5 * SPACE_SIZE"
+              [attr.y]="0.5 * SPACE_SIZE"
+              [ngClass]="getClockCircleClasses()">{{ countDown.get() }}</text>
+        <polyline *ngIf="countDown.isAbsent()"
+                  class="base"
+                  [attr.points]="clockNeedlesPoints"
+                  [ngClass]="getClockCircleClasses()"
+                  stroke-linecap="round" />
+    </g>
+    <line id="canal"
+          [attr.x1]="0"              [attr.y1]="(4 * SPACE_SIZE) + 8"
+          [attr.x2]="4 * SPACE_SIZE" [attr.y2]="(4 * SPACE_SIZE) + 8"
+          [ngClass]="'player' + getCurrentPlayer().value"
+          [attr.stroke-width]="2 * STROKE_WIDTH" />
+    <!-- Font Awesome Pro 5.15.4 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license (Commercial License) -->
+    <g id="modeCogView"
+       [attr.transform]="configViewTranslation">
+        <g id="modeCog"
+           (click)="onModeCogClick()">
+            <rect [attr.x]="0"
+                  [attr.y]="0"
+                  [attr.width]="SPACE_SIZE"
+                  [attr.height]="SPACE_SIZE"
+                  fill="none" />
+            <path [ngClass]="'player' + getCurrentPlayer().value"
+                  [attr.transform]="configCogTransformation"
+                  d="M487.4 315.7l-42.6-24.6c4.3-23.2 4.3-47 0-70.2l42.6-24.6c4.9-2.8 7.1-8.6 5.5-14-11.1-35.6-30-67.8-54.7-94.6-3.8-4.1-10-5.1-14.8-2.3L380.8 110c-17.9-15.4-38.5-27.3-60.8-35.1V25.8c0-5.6-3.9-10.5-9.4-11.7-36.7-8.2-74.3-7.8-109.2 0-5.5 1.2-9.4 6.1-9.4 11.7V75c-22.2 7.9-42.8 19.8-60.8 35.1L88.7 85.5c-4.9-2.8-11-1.9-14.8 2.3-24.7 26.7-43.6 58.9-54.7 94.6-1.7 5.4.6 11.2 5.5 14L67.3 221c-4.3 23.2-4.3 47 0 70.2l-42.6 24.6c-4.9 2.8-7.1 8.6-5.5 14 11.1 35.6 30 67.8 54.7 94.6 3.8 4.1 10 5.1 14.8 2.3l42.6-24.6c17.9 15.4 38.5 27.3 60.8 35.1v49.2c0 5.6 3.9 10.5 9.4 11.7 36.7 8.2 74.3 7.8 109.2 0 5.5-1.2 9.4-6.1 9.4-11.7v-49.2c22.2-7.9 42.8-19.8 60.8-35.1l42.6 24.6c4.9 2.8 11 1.9 14.8-2.3 24.7-26.7 43.6-58.9 54.7-94.6 1.5-5.5-.7-11.3-5.6-14.1zM256 336c-44.1 0-80-35.9-80-80s35.9-80 80-80 80 35.9 80 80-35.9 80-80 80z"/>
+        </g>
+        <g *ngIf="displayModePannel"
+           id="modePannel">
+            <rect class="base"
+                  [attr.x]="0"
+                  [attr.y]="(- listOfStyles.length * SPACE_SIZE)"
+                  [attr.width]="1 * SPACE_SIZE"
+                  [attr.height]="listOfStyles.length * SPACE_SIZE"
+                  stroke-width="8" />
+            <ng-container *ngFor="let proposedStyle of listOfStyles; index as y;">
+                <g [id]="proposedStyle.name"
+                   app-martian-chess-queen
+                   (click)="chooseStyle(y)"
+                   [attr.transform]="'translate(0, ' + (-(y+1) * SPACE_SIZE) + ')'"
```

Comment:
hmm pourquoi on translate négativement ? :o

Reply:
car on for de base en haut, on connais le bas, pas le haut